### ISSUE TITLE
Environment.OSVersion should detect the OS through the browser when running on WebAssembly

### ISSUE THREAD

**Issue by danroth27** [mono/mono#13002]
It looks like `Environment.OSVersion` returns "Unix 1.0.0.0" when running on wasm. Perhaps it should instead reflect the value of [window.navigator.platform`](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorID/platform), or whatever is the most reliable way to detect the OS from a browser?

**Comment by danroth27** [comment]
@kjpou1

**Comment by kjpou1** [comment]
@danroth27 I may have jumped the gun on this one.  The [Environment.OSVersion](https://docs.microsoft.com/en-us/dotnet/api/system.environment.osversion?view=netframework-4.7.2) is actually made up of a [PlatformID enum](https://docs.microsoft.com/en-us/dotnet/api/system.platformid?view=netframework-4.7.2).    

Even if we did put some code in to try to parse the operating system the problem with this is that the [window.navigator.platform](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorID/platform) can be spoofed.  So it will not be reliable anyway.

**Comment by danroth27** [comment]
> Even if we did put some code in to try to parse the operating system the problem with this is that the [window.navigator.platform](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorID/platform) can be incorrect. So it will not be reliable anyway.

Yes, I am aware of this, but I just thought it would be more useful than always returning "Unix 1.0.0.0"

**Comment by NattyNarwhal** [comment]
See these Core issues: https://github.com/dotnet/corefx/issues/23989 https://github.com/dotnet/corefx/issues/30706

**Comment by kjpou1** [comment]
I totally agree and the reason why I jumped on it so fast.

**Comment by akoeplinger** [comment]
There's also https://github.com/dotnet/corefx/issues/27417 which is related.

**Comment by mkArtakMSFT** [comment]
This is also being discussed https://github.com/dotnet/runtime/issues/33328

**Comment by mkArtakMSFT** [comment]
@lewing do you expect to align Mono implementation with the decision in the above referenced issue in dotnet/runtime?
